File: ttps-spring-clinicamascotas/src/ttps/spring/model/Mascota.java
package ttps.spring.model;

import java.util.Date;
import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import org.springframework.stereotype.Component;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Component
public class Mascota {
	@Id @GeneratedValue(strategy=GenerationType.IDENTITY)
	private long id;
	private String nombre;
	private Date fechaNacimiento;
	private String especie;
	private String raza;
	private char sexo;
	private String color;
	private String senas;
	private String foto;
	@ManyToOne(optional = true)
	@JoinColumn(name="id_dueno")
	private Usuario dueno;
	@ManyToOne(optional = true)
	@JoinColumn(name="id_veterinario")
	private Usuario veterinario;
	@OneToMany(mappedBy="mascota")
	@JsonIgnore
	private List<Visita> visitas;
	
	public long getId() {
		return id;
	}
	public void setId(long id) {
		this.id = id;
	}
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public Date getFechaNacimiento() {
		return fechaNacimiento;
	}
	public void setFechaNacimiento(Date fechaNacimiento) {
		this.fechaNacimiento = fechaNacimiento;
	}
	public String getEspecie() {
		return especie;
	}
	public void setEspecie(String especie) {
		this.especie = especie;
	}
	public String getRaza() {
		return raza;
	}
	public void setRaza(String raza) {
		this.raza = raza;
	}
	public char getSexo() {
		return sexo;
	}
	public void setSexo(char sexo) {
		this.sexo = sexo;
	}
	public String getColor() {
		return color;
	}
	public void setColor(String color) {
		this.color = color;
	}
	public String getSenas() {
		return senas;
	}
	public void setSenas(String senas) {
		this.senas = senas;
	}
	public String getFoto() {
		return foto;
	}
	public void setFoto(String foto) {
		this.foto = foto;
	}
	public Usuario getDueno() {
		return dueno;
	}
	public void setDueno(Usuario dueno) {
		this.dueno = dueno;
	}
	public Usuario getVeterinario() {
		return veterinario;
	}
	public void setVeterinario(Usuario veterinario) {
		this.veterinario = veterinario;
	}
	public List<Visita> getVisitas() {
		return visitas;
	}
	public void setVisitas(List<Visita> visitas) {
		this.visitas = visitas;
	}
	@Override
	public boolean equals(Object o) {
		if (!(o instanceof Mascota)){
            return false;
        }
		Mascota temp = (Mascota)o;
		return this.id == temp.getId();
	}
	
}

File: HospitalMascotas/src/app/modelos/mascota.ts
import { Usuario } from './usuario';

export class Mascota {
    
    private nombre: string;
    private fechaNacimiento: string;
    private especie: string;
    private raza: string;
    private sexo: CharacterData;
    private color: string;
    private senas: string;
    private foto: string;
    private dueno: Usuario;
    private veterinario: Usuario;
    private visitas: string;

    
    constructor (nombre: string, fechaNacimiento: string, especie: string, raza: string, sexo: CharacterData, color:string, senas: string, foto: string, dueno: Usuario, veterinario: Usuario){
        this.nombre = nombre;
        this.fechaNacimiento = fechaNacimiento;
        this.especie = especie;
        this.raza = raza;
        this.sexo = sexo;
        this.color = color;
        this.senas = senas;
        this.foto = foto;
        this.dueno = dueno;
        this.veterinario = veterinario;
        this.visitas = null;
    }
    public getVeterinario(): Usuario {
        return this.veterinario;
    }

    public setVeterinario(value: Usuario) {
        this.veterinario = value;
    }

    public getVisitas(): string {
        return this.visitas;
    }

    public setVisitas(value: string) {
        this.visitas = value;
    }

    public getDueno(): Usuario {
        return this.dueno;
    }

    public setDueno(value: Usuario) {
        this.dueno = value;
    }
    public getFoto(): string {
        return this.foto;
    }

    public setFoto(value: string) {
        this.foto = value;
    }
    public getSenas(): string {
        return this.senas;
    }

    public setSenas(value: string) {
        this.senas = value;
    }
    public getColor(): string {
        return this.color;
    }

    public setColor(value: string) {
        this.color = value;
    }
    public getSexo(): CharacterData {
        return this.sexo;
    }

    public setSexo(value: CharacterData) {
        this.sexo = value;
    }
    public getRaza(): string {
        return this.raza;
    }

    public setRaza(value: string) {
        this.raza = value;
    }
    public getEspecie(): string {
        return this.especie;
    }

    public setEspecie(value: string) {
        this.especie = value;
    }
    public getFechaNacimiento(): string {
        return this.fechaNacimiento;
    }

    public setFechaNacimiento(value: string) {
        this.fechaNacimiento = value;
    }
    public getNombre(): string {
        return this.nombre;
    }

    public setNombre(value: string) {
        this.nombre = value;
    }


}
